Dockerfile

```diff
@@ -1,20 +1,23 @@
-FROM node:16-alpine@sha256:a9b9cb880fa429b0bea899cd3b1bc081ab7277cc97e6d2dcd84bd9753b2027e1
+FROM node:16-alpine@sha256:a9b9cb880fa429b0bea899cd3b1bc081ab7277cc97e6d2dcd84bd9753b2027e1 as test-env
 
 # Install bash
-RUN apk add --no-cache bash
-
-# Create working directory
-RUN mkdir -p /app
+RUN apk add --no-cache bash && mkdir -p /app
 WORKDIR /app
 
 # Install all Node dependencies
 COPY package.json /app/
 COPY package-lock.json /app/
-
 RUN npm ci
 
 # Copy source code
 COPY . /app/
 
-# Define default command
+# Perform tests
+RUN npm run test
 CMD npm test
+
+FROM alpine:edge
```

Comment:
Should we just split the `Dockerfile` into `Dockerfile.test` for the existing testing image, then use `Dockerfile` for building a runnable image with the script baked-in. 

Like this, we keep their concerns separated. Both should be near-trivial Dockerfiles, so it's easy for anyone to understand what's going on there.

Reply:
I'm always hesitating against splitting Dockerfiles and prefer multi-stage Dockerfiles. I'll update the pull request with a fixed multi-stage Dockerfile so that you can review again. If you still prefer dedicated Dockerfiles I can still split them.

Some reasons for my preferences to multi-stage Dockerfiles:
- Out-of-the-box tool support for the `Dockerfile` syntax usually isn't enabled for files with the `Dockerfile.anything` name
- A combined Dockerfile reminds developers to consider both use cases (tests and runtime) when changing anything
- A simple `docker build ...` performs both tests and runtime on the go, so it's more failsafe and simplifies scripts
- I think especially for near-trivial Dockerfiles a multi-stage Dockerfile doesn't overwhelm anyone

That said: you're the maintainer and I'll follow your preferred way of going forward :)

Reply:
Thanks for sharing your thoughts. I'm going to pull your commit into this branch, the changes are perfectly fine :)